GitCommands/RevisionReader.cs

```diff
@@ -511,34 +470,101 @@ public StringLineReader(string s)
 
                 if (endIndex == -1)
                 {
-                    return ReadToEnd(advance);
+                    // Consider this as an error: PeakToEnd() should be explicitly used
+                    return null;
                 }
 
-                if (advance)
-                {
-                    _index = endIndex + 1;
-                }
+                _index = endIndex + 1;
 
                 return pool is not null
                     ? pool.Intern(_s, startIndex, endIndex - startIndex)
                     : _s.Substring(startIndex, endIndex - startIndex);
             }
 
-            public string? ReadToEnd(bool advance = true)
+            public StringLineReader TrimStart()
```

Comment:
This should be leveraging .NET implementations instead of our custom ones (e.g. `ROS<char>`).
/cc: @drewnoakes

Reply:
I implemented this in RevisionReader, see branch `tmp/log-zero-0` (tests are not adapted)
However, it seem to be slower. linux repo loads in 6-7 seconds instead of 5-6 seconds.

This could be due to the adaptions I made to StringPool. This implementation could maybe be replaced with https://docs.microsoft.com/en-us/windows/communitytoolkit/high-performance/stringpool

Reply:
First of all I'd look at rewriting the parsing logic to use ROS<char>, and then benchmark it. Then would add `StringPool`, and benchmark it again.
I'm finding it difficult to believe that our current implementation is less performant than with the use of ROS, but I haven't studied the code deeply enough to be sure.

Reply:
This optimisation work is somewhat orthogonal to this change though.

Reply:
ROS was implemented - one reason this PR had so many feature creeps...